packages/@sanity/form-builder/src/inputs/SelectInput.tsx

```diff
@@ -26,15 +26,34 @@ const SelectInput = React.forwardRef(function SelectInput(
   const validation = markers.filter((marker) => marker.type === 'validation')
   const errors = validation.filter((marker) => marker.level === 'error')
 
+  const itemFromOptionValue = (optionValue) => {
+    const index = Number(optionValue)
+
+    return items[index]
+  }
+
+  const optionValueFromItem = (item) => {
+    return items.indexOf(item)
```

Comment:
I believe this should be modified to a string, and not return `-1`, e.g. `return String(items.indexOf(item) || 0)`

Reply:
We are using `-1` as the first value for the undefined element though since `0` is the first index.

Reply:
Made it return a string in 0c9c47f03

Reply:
Ah, clever! Yes, that makes sense 👍